himbaechel/uarch/gatemate/pack.cc

```diff
@@ -2170,6 +2170,347 @@ void GateMatePacker::pack_ram()
+            move_ram_o(&ci, ctx->idf("TX_8B10B_BYPASS_I[%d]", i));
+            move_ram_o(&ci, ctx->idf("TX_CHAR_IS_K_I[%d]", i));
+            move_ram_o(&ci, ctx->idf("TX_CHAR_DISPMODE_I[%d]", i));
+            move_ram_o(&ci, ctx->idf("TX_CHAR_DISPVAL_I[%d]", i));
+            move_ram_o(&ci, ctx->idf("RX_8B10B_BYPASS_I[%d]", i));
+            move_ram_o(&ci, ctx->idf("REGFILE_ADDR_I[%d]", i));
+            move_ram_i(&ci, ctx->idf("RX_NOT_IN_TABLE_O[%d]", i));
+            move_ram_i(&ci, ctx->idf("RX_CHAR_IS_COMMA_O[%d]", i));
+            move_ram_i(&ci, ctx->idf("RX_CHAR_IS_K_O[%d]", i));
+            move_ram_i(&ci, ctx->idf("RX_DISP_ERR_O[%d]", i));
+        }
+        for (int i = 0; i < 3; i++) {
+            move_ram_o(&ci, ctx->idf("TX_PRBS_SEL_I[%d]", i));
+            move_ram_o(&ci, ctx->idf("LOOPBACK_I[%d]", i));
+            move_ram_o(&ci, ctx->idf("RX_PRBS_SEL_I[%d]", i));
+        }
+        move_ram_o(&ci, id_TX_RESET_I);
+        move_ram_o(&ci, id_TX_PCS_RESET_I);
+        move_ram_o(&ci, id_TX_PMA_RESET_I);
+        move_ram_o(&ci, id_PLL_RESET_I);
+        move_ram_o(&ci, id_TX_POWER_DOWN_N_I);
+        move_ram_o(&ci, id_TX_POLARITY_I);
+        move_ram_o(&ci, id_TX_PRBS_FORCE_ERR_I);
+        move_ram_o(&ci, id_TX_8B10B_EN_I);
+        move_ram_o(&ci, id_TX_ELEC_IDLE_I);
+        move_ram_o(&ci, id_TX_DETECT_RX_I);
+        move_ram_o(&ci, id_TX_CLK_I);
+        move_ram_o(&ci, id_RX_CLK_I);
+        move_ram_o(&ci, id_RX_RESET_I);
+        move_ram_o(&ci, id_RX_PMA_RESET_I);
+        move_ram_o(&ci, id_RX_EQA_RESET_I);
+        move_ram_o(&ci, id_RX_CDR_RESET_I);
+        move_ram_o(&ci, id_RX_PCS_RESET_I);
+        move_ram_o(&ci, id_RX_BUF_RESET_I);
+        move_ram_o(&ci, id_RX_POWER_DOWN_N_I);
+        move_ram_o(&ci, id_RX_POLARITY_I);
+        move_ram_o(&ci, id_RX_PRBS_CNT_RESET_I);
+        move_ram_o(&ci, id_RX_8B10B_EN_I);
+        move_ram_o(&ci, id_RX_EN_EI_DETECTOR_I);
+        move_ram_o(&ci, id_RX_COMMA_DETECT_EN_I);
+        move_ram_o(&ci, id_RX_SLIDE_I);
+        move_ram_o(&ci, id_RX_MCOMMA_ALIGN_I);
+        move_ram_o(&ci, id_RX_PCOMMA_ALIGN_I);
+        move_ram_o(&ci, id_REGFILE_CLK_I);
+        move_ram_o(&ci, id_REGFILE_WE_I);
+        move_ram_o(&ci, id_REGFILE_EN_I);
+        move_ram_i(&ci, id_TX_DETECT_RX_DONE_O);
+        move_ram_i(&ci, id_TX_DETECT_RX_PRESENT_O);
+        move_ram_i(&ci, id_TX_BUF_ERR_O);
+        move_ram_i(&ci, id_TX_RESET_DONE_O);
+        move_ram_i(&ci, id_RX_PRBS_ERR_O);
+        move_ram_i(&ci, id_RX_BUF_ERR_O);
+        move_ram_i(&ci, id_RX_BYTE_IS_ALIGNED_O);
+        move_ram_i(&ci, id_RX_BYTE_REALIGN_O);
+        move_ram_i(&ci, id_RX_RESET_DONE_O);
+        move_ram_i(&ci, id_RX_EI_EN_O);
+        move_ram_i(&ci, id_RX_CLK_O);
+        move_ram_i(&ci, id_PLL_CLK_O);
+        move_ram_i(&ci, id_REGFILE_RDY_O);
+
+        ci.params[id_RX_BUF_RESET_TIME] = Property(int_or_default(ci.params, id_RX_BUF_RESET_TIME, 3), 5);
```

Comment:
maybe it'd be nicer to have a static const array somewhere of all these parameters and their widths/defaults, rather than all this imperative code